diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/TrailingCommaRule.kt

```diff
@@ -0,0 +1,114 @@
+
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.ast.ElementType.COLLECTION_LITERAL_EXPRESSION
+import com.pinterest.ktlint.core.ast.ElementType.COMMA
+import com.pinterest.ktlint.core.ast.ElementType.DESTRUCTURING_DECLARATION
+import com.pinterest.ktlint.core.ast.ElementType.DESTRUCTURING_DECLARATION_ENTRY
+import com.pinterest.ktlint.core.ast.ElementType.INDICES
+import com.pinterest.ktlint.core.ast.ElementType.REFERENCE_EXPRESSION
+import com.pinterest.ktlint.core.ast.ElementType.STRING_TEMPLATE
+import com.pinterest.ktlint.core.ast.ElementType.TYPE_ARGUMENT_LIST
+import com.pinterest.ktlint.core.ast.ElementType.TYPE_PARAMETER
+import com.pinterest.ktlint.core.ast.ElementType.TYPE_PARAMETER_LIST
+import com.pinterest.ktlint.core.ast.ElementType.TYPE_PROJECTION
+import com.pinterest.ktlint.core.ast.ElementType.VALUE_ARGUMENT
+import com.pinterest.ktlint.core.ast.ElementType.VALUE_ARGUMENT_LIST
+import com.pinterest.ktlint.core.ast.ElementType.VALUE_PARAMETER
+import com.pinterest.ktlint.core.ast.ElementType.VALUE_PARAMETER_LIST
+import com.pinterest.ktlint.core.ast.ElementType.WHEN_CONDITION_WITH_EXPRESSION
+import com.pinterest.ktlint.core.ast.ElementType.WHEN_ENTRY
+import com.pinterest.ktlint.core.ast.children
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.jetbrains.kotlin.psi.psiUtil.siblings
+import org.jetbrains.kotlin.utils.addToStdlib.ifNotEmpty
+
+/**
+ * [1] Enumerations (In another rule)
+ * [2] Value arguments
+ * [3] Class properties and parameters
+ * [4] Function value parameters
+ * [5] Parameters with optional type (including setters)
+ * [6] Indexing suffix
+ * [7] Lambda parameters
+ * [8] when entry
+ * [9] Collection literals (in annotations)Type arguments
+ * [10] Type arguments
+ * [11] Type parameters
+ * [12] Destructuring declarations
+ */
+class TrailingCommaRule(private val configRules: List<RulesConfig>) : Rule("trailing-comma") {
+    private var isFixMode: Boolean = false
+    private val configuration by lazy {
+        TrailingCommaConfiguration(
+            this.configRules.getRuleConfig(TRAILING_COMMA)?.configuration ?: emptyMap()
+        )
+    }
+    private lateinit var emitWarn: EmitType
+
+    override fun visit(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: EmitType
+    ) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        val commonConfig by configRules.getCommonConfiguration()
+        if (commonConfig.kotlinVersion >= ktVersion) {
+            val (type, config) = when (node.elementType) {
+                VALUE_ARGUMENT_LIST -> Pair(VALUE_ARGUMENT, configuration.getParam("valueArgument"))
+                VALUE_PARAMETER_LIST -> Pair(VALUE_PARAMETER, configuration.getParam("valueParameter"))
+                INDICES -> Pair(REFERENCE_EXPRESSION, configuration.getParam("referenceExpression"))
```

Comment:
I think it should be called `indices` in confiuration too, now it's too confusing.

Reply:
Fixed